### PR TITLE
Upgrade roboletric and junit versions to be consistent

### PR DESCRIPTION
**Pull request by josephyanks**
# Why

The bumping of robolectric to 4.10 means that we no longer need jetifier to translate old support library dependencies.

# How

* Updates `robolectric` to `4.10` and `junit` to `4.13.2`
* Updates test dependencies in `expo-modules-test-core` to their latest versions
* A minor change needed to be made to the `expo-clipboard` test because of changes to the way that robolectric handles decoding bitmaps for tests.

# Test Plan

Ensured [unit tests still pass](https://github.com/josephyanks/expo/actions/runs/4896200135/jobs/8742724860)

Trying to verify end to end tests locally.

# Checklist

- [ ] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [ ] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: packages/expo-application/CHANGELOG.md
Comment (Kudo) [inline packages/expo-application/CHANGELOG.md:6]:
```suggestion

### 📚 3rd party library updates

- Updated `robolectric` to `4.10` and `junit` to `4.13.2`.  ([#22395](https://github.com/expo/expo/pull/22395) by [@josephyanks](https://github.com/josephyanks))

```

thanks for the great pr! most changes look good to me. it's just one nit for changelog. would you mind to update all changelog under a section like `3rd party library updates`?

Comment (josephyanks) [reply packages/expo-application/CHANGELOG.md:8]: Updated them all, thanks

**Comment by josephyanks** [comment]
@Kudo I'm not sure how much I need to worry about these other CI steps failing - it seems like every time they run they fail for a different reason. I've verified that the android unit tests pass locally, and can't get the android instrumentation tests to run. Let me know what needs to be done to get this merged, thanks!

**Comment by josephyanks** [comment]
@Kudo no worries, just want to make sure I'm being thorough! it should be good to merge once I get the approval then

File: packages/expo-clipboard/android/src/test/java/expo/modules/clipboard/ClipboardImageTest.kt
Comment (Kudo) [inline packages/expo-clipboard/android/src/test/java/expo/modules/clipboard/ClipboardImageTest.kt:160]: i love this comment. it makes things so clear. thanks for adding this.

**Review by Kudo — COMMENTED** [review]
last thing before approving and merging. i think updating bundles inside expo-dev-launcher are unrelated to this pr. i know that this might to resolve ci issues, but don't worry i'll try to fix ci issues in a separated pr. please help to revert the expo-dev-launcher bundles changes. thank you!

**Comment by josephyanks** [comment]
@Kudo Those two bundle updates from `expo-dev-launcher` should be properly reverted now. Thank you for reviewing!

**Review by Kudo — APPROVED** [review]
everything looks pretty in this pr, i'm going to merge this. thanks for the fantastic pr. 🔥

**Pull request merged**
